bottlecap/src/lifecycle/invocation/processor.rs

```diff
@@ -324,10 +324,33 @@ impl Processor {
         trace_processor: Arc<dyn trace_processor::TraceProcessor + Send + Sync>,
         trace_agent_tx: Sender<SendData>,
     ) {
+        let context = self.enrich_spans_at_paltform_done(request_id, status);
+
+        if self.tracer_detected {
+            if let Some(ctx) = context {
+                if ctx.invocation_span.trace_id != 0 && ctx.invocation_span.span_id != 0 {
+                    self.send_extension_spans(
+                        &tags_provider,
+                        &trace_processor,
+                        &trace_agent_tx,
+                        ctx,
+                    )
+                    .await;
+                }
+            }
+        }
+    }
+
```

Comment:
Too many ifs, can we simplify it?

```suggestion
        if self.tracer_detected {
            if let Some(context) = self.enrich_spans_at_platform_done(request_id, status) {
                if context.invocation_span.trace_id != 0 && context.invocation_span.span_id != 0 {
                    self.send_extension_spans(
                        &tags_provider,
                        &trace_processor,
                        &trace_agent_tx,
                        context,
                    )
                    .await;
                }
            }
        }
    }
```

Or something like 
```
let Some(context) = ... else { ... } ;
if self.tracer_detected && (trace_id and span_id) {
   ...
}
```

Reply:
Not sure if there is an easier way to simplify this. The if let cannot be inline with other ifs, so either a match, an indentation or an early return are needed.
Since I already minimized this method to be very small and simply, I don't see the need to simplify it